contact-center/ios-voice/VonageSDKClientVOIPExample/Controllers/Calls/CallController+CXProviderDelegate.swift

```diff
@@ -84,6 +84,31 @@ extension VonageCallController: CXProviderDelegate {
     func provider(_ provider: CXProvider, didDeactivate audioSession: AVAudioSession){
         VGVoiceClient.disableAudio(audioSession)
     }
+    func provider(_ provider: CXProvider, perform action: CXSetHeldCallAction){
+        guard let _ = self.vonageActiveCalls.value[action.callUUID]  else {
+            action.fail()
+            return
+        }
+        action.fulfill()
+        if (!action.isOnHold) {
+            self.client.unmute(action.callUUID.toVGCallID()) { err in
+                // TODO:
```

Comment:
What does Apple recommend here if there was an error @salvodicara ?

Reply:
`action.fail` should be invoked, but that can only be done synchronously...